scripts/update_compatibility_tests.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python3
+"""
+The following script takes care of adding new/removing versions or 
+replacing a version in the compatibility-test-matrices JSON files.
+
+To use this script, you'll need to have Python 3.9+ installed.
+
+Invocation:
+
+By default, the script assumes that adding a new version is the desired operation.
+Furthermore, it assumes that the compatibility-test-matrices directory is located
+in the .github directory and the script is invoked from the root of the repository.
+
+If any of the above is not true, you can use the '--type' and '--directory' flags
+to specify the operation and the directory respectively.
+
+Typically, an invocation would look like:
+
+    scripts/update_compatibility_tests.py --recent_version v4.3.0 --new_version v4.4.0
+
+The three operations currently added are:
+
+ - ADD: Add a new version to the JSON files. Requires both '--recent_version' and
+        '--new_version' options to be set.
+ - REPLACE: Replace an existing version with a new one. Requires both '--recent_version'
+            and '--new_version' options to be set.
+ - REMOVE: Remove an existing version from the JSON files. Requires only the
+           '--recent_version' options to be set.
+
+For more information, use the '--help' flag to see the available options.    
+"""
+import argparse
+import os
+import json
+import enum
+from collections import defaultdict
+from typing import Tuple, Generator, Optional, Dict, List, Any
+
+# Directory to operate in
+DIRECTORY: str = ".github/compatibility-test-matrices"
+# JSON keys to search in.
+KEYS: Tuple[str, str] = ("chain-a", "chain-b")
+# Toggle if required. Indent = 2 matches our current formatting.
+DUMP_ARGS: Dict[Any, Any] = {
+    "indent": 2,
+    "sort_keys": False,
+    "ensure_ascii": False,
+}
+# Suggestions for recent and new versions.
+SUGGESTION: str = "for example, v4.3.0 or 4.3.0"
+# Supported Operations.
+Operation = enum.Enum("Operation", ["ADD", "REMOVE", "REPLACE"])
+
+
+def find_json_files(
+    directory: str, ignores: Tuple[str] = (".",)
+) -> Generator[str, None, None]:
+    """Find JSON files in a directory. By default, ignore hidden directories."""
+    for root, dirs, files in os.walk(directory):
+        dirs[:] = (d for d in dirs if not d.startswith(ignores))
+        for file_ in files:
+            if file_.endswith(".json"):
+                yield os.path.join(root, file_)
+
+
+def has_release_version(json_file: Any, keys: Tuple[str, str], version: str) -> bool:
+    """Check if the json file has the version in question."""
+    rows = (json_file[key] for key in keys)
+    return any(version in row for row in rows)
+
+
+def sorter(key: str) -> str:
+    """Since 'main' < 'vX.X.X' and we want to have 'main' as the first entry
+    in the list, we return a version that is considerably large. If ibc-go
+    reaches this version I'll wear my dunce hat and go sit in the corner.
```

Comment:
I will also eat my shorts on live stream

Reply:
At our current pace of 3 major releases a year we can get there in 33330 years. Think about the poor person on whose face this time bomb will explode, man.

Reply:
a quantum commit in the year 35353 "F*** you, DimitrisJim."